### ISSUE TITLE
@psalm-suppress with misspelled/non-existing issue

### ISSUE THREAD

**Issue by caugner** [vimeo/psalm#6120]
I would expect that the following causes an error:

```php
/**
 * @psalm-suppress NonExistingIssue
 */
```

https://psalm.dev/r/621dbf177f

**Comment by bdsl** [comment]
@caugner I'm not sure. Doesn't allowing `@psalm-suppress NonExistingIssue` provide a useful upgrade path, for upgrading Psalm to a new version that can detect `NonExistingIssue` in a large codebase.

If you don't want to use baselining and you can't fix all the `NonExistingIssue`s at once you'd need to suppress them. If there are many and you're doing trunk based development and wanting to do small commits you might want to first test out the new version of Psalm to find the issues locally, and then suppress the issues in batches, pushing to CI at each step, and only commit the Psalm upgrade later.

**Comment by caugner** [comment]
In a large codebase, you will definitely want to use a baseline, or at least suppress issues globally or per directory. I would recommend to create a branch, update psalm there, and add all the required `@psalm-suppress` until CI passes, while continuously rebasing on your "trunk". For that scenario, you will more likely want a `psalm --alter` command that just adds all those `@psalm-suppress` annotations automatically. But baselining is still way better, because it doesn't pollute the VCS history for all those affected files. For a *very* large codebase, you will likely want different configs per module/package.

Note that Psalm config has a `findUnusedPsalmSuppress` option, and even if that is enabled, Psalm doesn't find `@psalm-suppress` annotations with invalid issue names.

**Comment by weirdan** [comment]
I don't think this can be done currently as those unknown issues may refer to descendants of `PluginIssue`, provided by plugins. Currently Psalm doesn't have full list of issues that could be emitted. Fixing that would require plugins to register all custom issues, so it can only be done in major version.

**Comment by bdsl** [comment]
I don't have a strong feeling about this - it's hard to imagine a scenario where a new psalm version introduces so many new issues that it would be really uncomfortable to keep a branch open long enough to add all the suppress tags, even if doing trunk based. And I agree adding them to the baseline is potentially better than adding `@suppress` tags - even more so if https://github.com/vimeo/psalm/issues/5831 is implemented.
